src/frontends/tensorflow/src/op/concat.cpp

```diff
@@ -15,36 +15,52 @@ namespace ov {
 namespace frontend {
 namespace tensorflow {
 namespace op {
+OutputVector translate_concat_base(const NodeContext& node, const OutputVector& inputs, int64_t axis) {
+    auto concat = make_shared<Concat>(inputs, axis);
+    set_node_name(node.get_name(), concat);
+    return {concat};
+}
 
 OutputVector translate_concat_op(const NodeContext& node) {
-    size_t axis_idx, concat_idx_start, concat_idx_stop;
-    if (node.get_op_type() == "ConcatV2") {
-        axis_idx = node.get_input_size() - 1;
-        concat_idx_start = 0;
-        concat_idx_stop = node.get_input_size() - 1;
-    } else if (node.get_op_type() == "Concat") {
-        axis_idx = 0;
-        concat_idx_start = 1;
-        concat_idx_stop = node.get_input_size();
-    } else {
-        TENSORFLOW_OP_VALIDATION(node, false, "Incorrect operation type.");
-    }
+    // The difference between Concat and ConcatV2 is that
+    // axis is the first input for Concat
+    // and is the last input to ConcatV2
+    default_op_checks(node, 2, {"Concat", "ConcatV2"});
+    auto input_size = node.get_input_size();
 
-    std::vector<int64_t> tf_concat_axis_vec;
-    get_const_input(node, axis_idx, &tf_concat_axis_vec);
-    int64_t concat_axis = tf_concat_axis_vec[0];
+    int64_t axis;
+    OutputVector inputs;
 
-    OutputVector ng_args;
-    for (size_t i = concat_idx_start; i < concat_idx_stop; i++) {
-        Output<Node> ng_arg = node.get_input(static_cast<int>(i));
-        ng_args.push_back(ng_arg);
+    if (node.get_op_type() == "Concat") {
+        std::vector<int64_t> axis_vector;
+        get_const_input(node, 0, &axis_vector);
+        TENSORFLOW_OP_VALIDATION(node,
+                                 axis_vector.size() > 0,
+                                 "Input model is incorrect: axis input for Concat operation must not be empty.");
+        axis = axis_vector[0];
```

Comment:
What if `axis_vector` is longer than 1 element? Is it possible? If we not expect it I suggest having validation check that `axis_vector` is single element vector.